src/hooks/sync-sites/sync-sites-context.tsx

```diff
@@ -99,6 +108,79 @@ export function SyncSitesProvider( { children }: { children: React.ReactNode } )
 	const { syncSites, isFetching, refetchSites } = useSyncSitesData();
 	useListenDeepLinkConnection( { refetchSites } );
 
+	const { client } = useAuth();
+	const { pushStatesProgressInfo } = useSyncStatesProgressInfo();
+	const hasInitialized = useRef( false );
```

Comment:
What if we use a regular variable instead of `useRef`?

Reply:
Thanks for the explanation. I think it's a good approach. I'm extra suspicious of states being saved in useRef, but it makes sense for this use case.